librz/core/cmd_analysis.c

```diff
@@ -8861,7 +8861,25 @@ RZ_IPI int rz_cmd_analysis(void *data, const char *input) {
 	case 'i': cmd_analysis_info(core, input + 1); break; // "ai"
 	case 'r': cmd_analysis_reg(core, input + 1); break; // "ar"
 	case 'e': cmd_analysis_esil(core, input + 1); break; // "ae"
-	case 'L': return rz_core_cmd0(core, "e asm.arch=??"); break;
+	case 'L': { // aL
+		RzCmdStateOutput state = { 0 };
+		switch (input[1]) {
+		case 'j': {
+			state.mode = RZ_OUTPUT_MODE_JSON;
+			state.d.pj = pj_new();
+			break;
+		}
+		case 'q': {
+			state.mode = RZ_OUTPUT_MODE_QUIET;
+			break;
+		}
+		default: {
+			state.mode = RZ_OUTPUT_MODE_STANDARD;
+			break;
+		}
+		}
+		rz_core_asm_plugins_print(core, NULL, &state);
```

Comment:
after this, you should put something like https://github.com/rizinorg/rizin/blob/dev/librz/core/cmd_api.c#L657 . Otherwise JSON mode is not printed. That is because when converting commands to newshell, these things will be done automatically.